src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -860,40 +876,46 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
 
         // Discover translation directories
         $dirs = array();
-        if (class_exists('Symfony\Component\Validator\Validation')) {
-            $r = new \ReflectionClass('Symfony\Component\Validator\Validation');
+        if (class_exists(Validation::class)) {
+            $r = new \ReflectionClass(Validation::class);
 
             $dirs[] = dirname($r->getFileName()).'/Resources/translations';
         }
-        if (class_exists('Symfony\Component\Form\Form')) {
-            $r = new \ReflectionClass('Symfony\Component\Form\Form');
+        if (class_exists(Form::class)) {
+            $r = new \ReflectionClass(Form::class);
 
             $dirs[] = dirname($r->getFileName()).'/Resources/translations';
         }
-        if (class_exists('Symfony\Component\Security\Core\Exception\AuthenticationException')) {
-            $r = new \ReflectionClass('Symfony\Component\Security\Core\Exception\AuthenticationException');
+
+        $container->addResource(new ClassExistenceResource(AuthenticationException::class));
+        if (class_exists(AuthenticationException::class)) {
+            $r = new \ReflectionClass(AuthenticationException::class);
 
             $dirs[] = dirname(dirname($r->getFileName())).'/Resources/translations';
         }
         $rootDir = $container->getParameter('kernel.root_dir');
         foreach ($container->getParameter('kernel.bundles_metadata') as $name => $bundle) {
-            if (is_dir($dir = $bundle['path'].'/Resources/translations')) {
+            $container->addResource(new FileExistenceResource($dir = $bundle['path'].'/Resources/translations'));
+            if (is_dir($dir)) {
                 $dirs[] = $dir;
             }
-            if (is_dir($dir = $rootDir.sprintf('/Resources/%s/translations', $name))) {
+            $container->addResource(new FileExistenceResource($dir = $rootDir.sprintf('/Resources/%s/translations', $name)));
+            if (is_dir($dir)) {
                 $dirs[] = $dir;
             }
         }
 
         foreach ($config['paths'] as $dir) {
+            $container->addResource(new FileExistenceResource($dir));
             if (is_dir($dir)) {
                 $dirs[] = $dir;
             } else {
                 throw new \UnexpectedValueException(sprintf('%s defined in translator.paths does not exist or is not a directory', $dir));
             }
         }
 
-        if (is_dir($dir = $rootDir.'/Resources/translations')) {
+        $container->addResource(new FileExistenceResource($dir = $rootDir.'/Resources/translations'));
```

Comment:
maybe not when the dir is not found in fact - then should be added in an "else" case of the following "if", isn't it?